app/rdf/query-latest-cube-iri.ts

```diff
@@ -0,0 +1,44 @@
+/** Creates SPARQL query to fetch latest cube iri.
+ * Works for both versioned and unversioned cubes.
+ */
+export const getLatestCubeIriQuery = (cubeIri: string) => {
+  return `PREFIX cube: <https://cube.link/>
+PREFIX schema: <http://schema.org/>
+
+SELECT ?iri WHERE {
+  {
+    # Versioned cube.
+    SELECT ?iri WHERE {
+      VALUES ?oldIri { <${cubeIri}> }
+      ?versionHistory schema:hasPart ?oldIri .
+      ?versionHistory schema:hasPart ?iri .
+      ?iri schema:version ?version .
+      ?iri schema:creativeWorkStatus ?status .
+      ?oldIri schema:creativeWorkStatus ?oldStatus .
+      FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status IN (?oldStatus, <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>))
+    }
+  } UNION {
+    {
+      # Non-versioned cube.
+      SELECT ?iri WHERE {
+        VALUES ?iri { <${cubeIri}> }
+        ?iri cube:observationConstraint ?shape .
+        ?iri schema:creativeWorkStatus ?status .
+        FILTER(NOT EXISTS { ?iri schema:expires ?expires . } && ?status = <https://ld.admin.ch/vocabulary/CreativeWorkStatus/Published>)      
+      }
```

Comment:
this subquery would imply to show non-versioned cubes only if they would be Published... (line 27)
this condition may be relaxed to show draft cubes of any status.

... then this will also require an additional `FILTER NOT EXISTS { ?versionHistory schema:hasPart ?iri . } ` to mutually exclusive with the versioned cube query and not interfere by adding draft cubes to the verioned cube case.

Reply:
Yes, this would be the case. I think it's something I would expect – if there's a link to an unversioned cube version and somebody clicks it, they would open a dataset preview page. By default we don't show draft datasets in the search but require an explicit action to enable them and this logic would not match that behavior 👀

As we don't have a way to specify status of a cube in the URL, I think it makes sense to default to published cubes, especially in Visualize PROD. What do you think? Otherwise we would need to implement a way to keep cube status in the URL.

Reply:
non-versioned cubes could also be Draft (initially) especially on LINDAS TEST or INT. 
it won't hurt to able to display them (once a user clicked on those links)